hal/timer/periph.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 The Embedded Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package timer
+
+import (
+	"unsafe"
+
+	"github.com/embeddedgo/kendryte/hal/internal"
+	"github.com/embeddedgo/kendryte/p/mmap"
+	"github.com/embeddedgo/kendryte/p/sysctl"
+	"github.com/embeddedgo/kendryte/p/timer"
+)
+
+type Periph struct {
+	*timer.Periph
```

Comment:
Don't use timer.Periph here. Copy/paste the timer.Periph fields, make them unexported and replace Rxxx types with mmio.U32.

Packages in p directory document peripherals in more readable way than SVD. They are also intended to play with registers directly. But they are sometimes huge because of a different type and a set of methods for every register. The intention was to avoid accidental use of bits from wrong register.

But we don't want hal/timer to depend on p/timer. The number of types and methods in p/timer needlessly pollute the runtime.pclntab section.